### ISSUE TITLE
RFE: allow clamping username and permissions for source RPMs

### ISSUE THREAD

**Issue by NeilHanlon** [rpm-software-management/rpm#2604]
when building a source RPM, the username and filesystem permissions from the build host are preserved and end up propagating to a machine you extract them onto, e.g.. To be able to reproduce SRPMs, we should have predictable conventions for the permissions and ownership of files.

The idea we have come up with in the reproducibility hackfest was to adopt a git-style model of permission storage, i.e., dropping most information about the unix permissions. This model also just makes more sense, overall, for what source RPMS are.

Additionally, it would be useful to clamp the ownership of the files to root:root--though this will necessitate ensuring that the applications which work with RPM input/output respect this clamping and change the permissions if a user extracts or installs it. (Namely, we don't want a user to install an RPM with files they cannot touch or see)

**Comment by pmatilai** [comment]
Indeed, storing the local user/group into src.rpm makes no sense whatsoever. It's an anachronism from the ancient days before Buildroot was a thing that has somehow survived all the way up to this date. We've made some effort to swipe the symptoms under the carpet (ie avoid pointless warnings when this happens) but might as well address the root cause instead.

**Comment by keszybz** [comment]
> it would be useful to clamp the ownership of the files to root:root--though this will necessitate ensuring that the applications which work with RPM input/output respect this clamping and change the permissions if a user extracts or installs it. (Namely, we don't want a user to install an RPM with files they cannot touch or see)

Actually this is not an issue. A user cannot create files they cannot touch or see: if we do the proposed change, if running as root, the file is created owned by root, but anyway, for root the ownership and permission mask are ignored. If running as a user, they cannot create any files not owned by them.

So please just store the files as `uid=0`, `gid=0`, `mask=-rwxrwxraw` or `drwxrwxrwx`. This way, the right thing will happen, i.e. file will always be owned by the user and the permission mask will be determined by the user's umask.

**Comment by pmatilai** [comment]
Yes, the only real change needed here is the stored user/group. I doubt it *ever* made sense, really
